capsules/src/seven_segment.rs

```diff
@@ -0,0 +1,371 @@
+//! Provides userspace access to 7 segment digit displays.
+//!
```

Comment:
Can you add what hardware this was developed with?

Reply:
Done

Reply:
Is there some sort of switch? How is the digit selected?

Reply:
There is one pin for each digit. The buffer I used is an array containing NUM_DIGITS elements, each element being a pattern corresponding to one digit. The driver goes through this array constantly, to represent each digit one after the other (but appearing simultaneous to the human eye), similarly to how the led matrix driver for the Microbit works. So there isn't necessarily one digit selected at a time, rather every digit is in use successively. If one of the elements in the buffer remains unchanged (0b00000000), that means that none of the segments on that specific digit are on, but the driver will go over that element anyway and represent nothing every time it loops back, until the buffer is changed.

Reply:
Is this a specific product you can link to, or tutorial? How would someone else use this driver?

Reply:
I added the pin connections in the documentation: https://github.com/WyliodrinEmbeddedIoT/tock/blob/f36f2c983e5aa9353aa562ef1e750ebe16496068/capsules/src/seven_segment.rs#L12

Reply:
Do you have a part number for the display? Capsules generally reference specific parts that they were initially tested with.

Reply:
I added the part numbers in the documentation.